K9 mail k9-6.300 Crash then email openpgp

### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

6.300

### Where did you get the app from?

Other

### Android version

12

### Device model

Pixel 6 pro

### Steps to reproduce

I open k9 mail I click on an encrypted message with openpgp Key yubico I carry out the decript procedure I enter PIN I touch yubico golden button from ok the drop-down menu and then k9 mail crashes

### Expected behavior

Decrypt email

### Actual behavior

Crash app

![Screenshot_20220804-225238](https://user-images.githubusercontent.com/35575744/182950553-68f84243-c673-4321-af90-6980c13da2ba.png)


### Logs

_No response_

here is my debug file

`--------- beginning of crash
08-05 07:08:30.332 27132 27132 E AndroidRuntime: FATAL EXCEPTION: main
08-05 07:08:30.332 27132 27132 E AndroidRuntime: Process: com.fsck.k9, PID: 27132
08-05 07:08:30.332 27132 27132 E AndroidRuntime: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=33148, result=-1, data=Intent { act=org.openintents.openpgp.action.DECRYPT_VERIFY (has extras) }} to activity {com.fsck.k9/com.fsck.k9.activity.MessageList}: java.lang.IllegalStateException: Check failed.
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:5368)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5407)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2253)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7870)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: Caused by: java.lang.IllegalStateException: Check failed.
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at com.fsck.k9.ui.messageview.MessageViewContainerFragment.getMessageViewFragment(MessageViewContainerFragment.kt:46)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at com.fsck.k9.ui.messageview.MessageViewContainerFragment.onPendingIntentResult(MessageViewContainerFragment.kt:224)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at com.fsck.k9.activity.MessageList.onActivityResult(MessageList.kt:1290)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.app.Activity.dispatchActivityResult(Activity.java:8388)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:5361)
08-05 07:08:30.332 27132 27132 E AndroidRuntime: 	... 13 more
--------- beginning of main
08-05 11:47:27.574 10717 10773 W com.fsck.k9: Accessing hidden method Lcom/android/org/conscrypt/ConscryptEngineSocket;->setHostname(Ljava/lang/String;)V (max-target-q,core-platform-api, reflection, denied)
08-05 11:47:27.574 10717 10773 W com.fsck.k9: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->setHostname(Ljava/lang/String;)V (max-target-q,core-platform-api, reflection, denied)
08-05 11:47:27.574 10717 10773 W com.fsck.k9: Accessing hidden method Lcom/android/org/conscrypt/AbstractConscryptSocket;->setHostname(Ljava/lang/String;)V (max-target-q, reflection, denied)
08-05 11:47:29.424 10717 10773 W com.fsck.k9: Accessing hidden method Lcom/android/org/conscrypt/ConscryptEngineSocket;->setHostname(Ljava/lang/String;)V (max-target-q,core-platform-api, reflection, denied)
08-05 11:47:29.424 10717 10773 W com.fsck.k9: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->setHostname(Ljava/lang/String;)V (max-target-q,core-platform-api, reflection, denied)
08-05 11:47:29.425 10717 10773 W com.fsck.k9: Accessing hidden method Lcom/android/org/conscrypt/AbstractConscryptSocket;->setHostname(Ljava/lang/String;)V (max-target-q, reflection, denied)
08-05 11:47:30.170 10717 10773 W com.fsck.k9: Accessing hidden method Lcom/android/org/conscrypt/ConscryptEngineSocket;->setHostname(Ljava/lang/String;)V (max-target-q,core-platform-api, reflection, denied)
08-05 11:47:30.171 10717 10773 W com.fsck.k9: Accessing hidden method Lcom/android/org/conscrypt/OpenSSLSocketImpl;->setHostname(Ljava/lang/String;)V (max-target-q,core-platform-api, reflection, denied)
08-05 11:47:30.171 10717 10773 W com.fsck.k9: Accessing hidden method Lcom/android/org/conscrypt/AbstractConscryptSocket;->setHostname(Ljava/lang/String;)V (max-target-q, reflection, denied)
08-05 11:47:32.455 10717 10826 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=6bb8607a-9988-4777-a996-b14437d90b32, tags={ com.fsck.k9.job.MailSyncWorker, MailSync } ]
08-05 11:47:36.585 10717 10810 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=7746d585-2ab6-4d69-acce-83977859b2b1, tags={ com.fsck.k9.job.MailSyncWorker, MailSync } ]
08-05 11:47:37.412 10717 10803 I WM-WorkerWrapper: Worker result SUCCESS for Work [ id=84f50899-d728-4a94-9098-3612c929d7f7, tags={ com.fsck.k9.job.MailSyncWorker, MailSync } ]
08-05 11:51:38.517 10717 13061 D mali_cmarp_predictor: checking cmar_predictor for com.fsck.k9
08-05 11:51:38.568 10717 10717 D CompatibilityChangeReporter: Compat change id reported: 171228096; UID 10323; state: ENABLED
08-05 11:51:42.283 10717 13061 D OpenGLRenderer: endAllActiveAnimators on 0xb40000787073c4e0 (ConstraintLayout) with handle 0xb4000077807828a0
08-05 11:51:46.435 10717 13061 D OpenGLRenderer: endAllActiveAnimators on 0xb4000078707fea30 (FrameLayout) with handle 0xb400007780746e70
08-05 11:51:52.877 10717 13091 I K9StorageEditor: Committing preference changes
08-05 11:51:52.885 10717 13091 I K9StorageEditor: Preferences commit took 8 ms
08-05 11:51:58.006 10717 13061 D OpenGLRenderer: endAllActiveAnimators on 0xb400007870838fa0 (RippleDrawable) with handle 0xb40000778078f9b0
08-05 11:51:59.018 10717 10773 I MessagingController: Running command 'clearNotifications', seq = 37 (foreground priority)
08-05 11:51:59.021 10717 10773 I MessagingController:  Command 'clearNotifications' completed
08-05 11:51:59.466 10717 13061 D OpenGLRenderer: endAllActiveAnimators on 0xb4000078707f5c90 (RippleDrawable) with handle 0xb40000778078f9b0
08-05 11:52:09.627 10717 10717 D MessageViewFragment: MessageViewFragment displaying message MessageReference(accountUuid=2506f814-1ec5-4632-b625-285e09997cd8, folderId=2, uid=18)
08-05 11:52:09.628 10717 10717 D MessageLoaderHelper: Creating new local message loader
08-05 11:52:09.653 10717 10773 I MessagingController: Running command 'markMessageAsOpened', seq = 38 (foreground priority)
08-05 11:52:09.653 10717 10773 I MessagingController:  Command 'markMessageAsOpened' completed
08-05 11:52:09.679 10717 10717 D MessageCryptoHelper: received progress status: 0 / 100
08-05 11:52:09.687 10717 10717 D MessageCryptoHelper: OpenPGP API decryptVerify result code: 2
08-05 11:52:15.956 10717 10717 D CompatibilityChangeReporter: Compat change id reported: 78294732; UID 10323; state: DISABLED
08-05 11:52:15.956 10717 10717 D AndroidRuntime: Shutting down VM
08-05 11:52:15.957 10717 10717 E AndroidRuntime: FATAL EXCEPTION: main
08-05 11:52:15.957 10717 10717 E AndroidRuntime: Process: com.fsck.k9, PID: 10717
08-05 11:52:15.957 10717 10717 E AndroidRuntime: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=33148, result=-1, data=Intent { act=org.openintents.openpgp.action.DECRYPT_VERIFY (has extras) }} to activity {com.fsck.k9/com.fsck.k9.activity.MessageList}: java.lang.IllegalStateException: Check failed.
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:5368)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5407)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2253)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:106)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.os.Looper.loopOnce(Looper.java:201)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:288)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:7870)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: Caused by: java.lang.IllegalStateException: Check failed.
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at com.fsck.k9.ui.messageview.MessageViewContainerFragment.getMessageViewFragment(MessageViewContainerFragment.kt:46)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at com.fsck.k9.ui.messageview.MessageViewContainerFragment.onPendingIntentResult(MessageViewContainerFragment.kt:224)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at com.fsck.k9.activity.MessageList.onActivityResult(MessageList.kt:1290)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.app.Activity.dispatchActivityResult(Activity.java:8388)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:5361)
08-05 11:52:15.957 10717 10717 E AndroidRuntime: 	... 13 more
08-05 11:52:15.970 10717 10717 I Process : Sending signal. PID: 10717 SIG: 9
08-05 11:52:24.460 13435 13435 E com.fsck.k9: Not starting debugger since process cannot load the jdwp agent.
08-05 11:52:24.469 13435 13435 D CompatibilityChangeReporter: Compat change id reported: 171979766; UID 10323; state: ENABLED
08-05 11:52:24.486 13435 13435 V GraphicsEnvironment: ANGLE Developer option for 'com.fsck.k9' set to: 'default'
08-05 11:52:24.487 13435 13435 V GraphicsEnvironment: ANGLE GameManagerService for com.fsck.k9: false
08-05 11:52:24.487 13435 13435 V GraphicsEnvironment: Neither updatable production driver nor prerelease driver is supported.
08-05 11:52:24.487 13435 13435 D NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: false
08-05 11:52:24.488 13435 13435 D NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: false
08-05 11:52:24.523 13435 13435 D CompatibilityChangeReporter: Compat change id reported: 160794467; UID 10323; state: ENABLED
08-05 11:52:24.529 13435 13435 V K9JobManager: scheduling all jobs
08-05 11:52:24.529 13435 13435 V K9JobManager: canceling mail sync job
08-05 11:52:24.529 13435 13435 V MailSyncWorkerManager: Scheduling mail sync worker for d4df3f48-52ee-40c4-8e80-905a725ff8e7
08-05 11:52:24.529 13435 13435 V MailSyncWorkerManager:   sync interval: 60 minutes
08-05 11:52:24.530 13435 13435 V MailSyncWorkerManager:   last sync time: ven ago 05 11:30:17 CEST 2022
08-05 11:52:24.530 13435 13435 V MailSyncWorkerManager:   initial delay: 2272950 ms
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager: Scheduling mail sync worker for eba8a6f0-e639-4750-8781-edcce1909a24
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager:   sync interval: 60 minutes
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager:   last sync time: ven ago 05 11:30:55 CEST 2022
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager:   initial delay: 2311324 ms
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager: Scheduling mail sync worker for 7eb6f056-5065-4ba0-a488-80d4025559a2
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager:   sync interval: 60 minutes
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager:   last sync time: ven ago 05 11:47:36 CEST 2022
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager:   initial delay: 3312043 ms
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager: Scheduling mail sync worker for 32db3369-7c3b-4732-aed9-ba742ed612c7
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager:   sync interval: 60 minutes
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager:   last sync time: ven ago 05 11:47:37 CEST 2022
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager:   initial delay: 3312867 ms
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager: Scheduling mail sync worker for 2506f814-1ec5-4632-b625-285e09997cd8
08-05 11:52:24.531 13435 13435 V MailSyncWorkerManager:   sync interval: 60 minutes
08-05 11:52:24.532 13435 13435 V MailSyncWorkerManager:   last sync time: ven ago 05 11:47:32 CEST 2022
08-05 11:52:24.532 13435 13435 V MailSyncWorkerManager:   initial delay: 3307904 ms
08-05 11:52:24.532 13435 13435 V MailSyncWorkerManager: Scheduling mail sync worker for f6e4e3b5-d88b-4048-a6f1-8037698f0879
08-05 11:52:24.532 13435 13435 V MailSyncWorkerManager:   sync interval: 60 minutes
08-05 11:52:24.532 13435 13435 V MailSyncWorkerManager:   last sync time: ven ago 05 11:30:18 CEST 2022
08-05 11:52:24.532 13435 13435 V MailSyncWorkerManager:   initial delay: 2273730 ms
08-05 11:52:24.532 13435 13435 V MailSyncWorkerManager: Scheduling mail sync worker for 1fd749e9-0f34-4f31-a097-ebf49fa9b570
08-05 11:52:24.532 13435 13435 V MailSyncWorkerManager:   sync interval: 60 minutes
08-05 11:52:24.532 13435 13435 V MailSyncWorkerManager:   last sync time: ven ago 05 11:30:57 CEST 2022
08-05 11:52:24.532 13435 13435 V MailSyncWorkerManager:   initial delay: 2312539 ms
08-05 11:52:24.533 13435 13435 V MessageProvider: Registering content resolver notifier
08-05 11:52:24.534 13435 13435 V SystemLocaleManager: Enable LocaleBroadcastReceiver
08-05 11:52:24.545 13435 13458 D mali_cmarp_predictor: checking cmar_predictor for com.fsck.k9
08-05 11:52:24.547 13435 13472 V PushController: PushController.initInBackground()
08-05 11:52:24.547 13435 13472 V PushController: PushController.updatePushers()
08-05 11:52:24.548 13435 13472 V PushController: ..Running PushControllers: []
08-05 11:52:24.548 13435 13472 V PushServiceManager: PushServiceManager.stop()
08-05 11:52:24.548 13435 13472 V PushServiceManager: ..PushService is not running
08-05 11:52:24.548 13435 13472 V BootCompleteManager: Disable BootCompleteReceiver
08-05 11:52:24.549 13435 13472 V PushController: PushController.updatePushers()
08-05 11:52:24.549 13435 13472 V PushController: ..Running PushControllers: []
08-05 11:52:24.549 13435 13472 V PushServiceManager: PushServiceManager.stop()
08-05 11:52:24.549 13435 13472 V PushServiceManager: ..PushService is not running
08-05 11:52:24.549 13435 13472 V BootCompleteManager: Disable BootCompleteReceiver
08-05 11:52:24.586 13435 13453 I MessagingController: Running command 'clearNotifications', seq = 0 (foreground priority)
08-05 11:52:24.588 13435 13453 I MessagingController:  Command 'clearNotifications' completed
08-05 11:52:24.589 13435 13435 D CompatibilityChangeReporter: Compat change id reported: 171228096; UID 10323; state: ENABLED
08-05 11:52:24.595 13435 13453 I MessagingController: Running command 'clearNotifications', seq = 1 (foreground priority)
08-05 11:52:24.598 13435 13453 I MessagingController:  Command 'clearNotifications' completed
08-05 11:52:27.658 13435 13458 D OpenGLRenderer: endAllActiveAnimators on 0xb40000787071d530 (ConstraintLayout) with handle 0xb400007780741920
08-05 11:52:46.412 13435 13458 D OpenGLRenderer: endAllActiveAnimators on 0xb40000787073f1f0 (FrameLayout) with handle 0xb400007780734cf0
08-05 11:52:49.284 13435 13458 D OpenGLRenderer: endAllActiveAnimators on 0xb4000078708375e0 (MenuPopupWindow$MenuDropDownListView) with handle 0xb40000778074fbd0
08-05 11:52:49.419 13435 13458 D OpenGLRenderer: endAllActiveAnimators on 0xb400007870818630 (RippleDrawable) with handle 0xb40000778072ba80
08-05 11:52:51.484 13435 13435 D CompatibilityChangeReporter: Compat change id reported: 78294732; UID 10323; state: DISABLED
08-05 11:52:51.486 13435 13481 V LogcatLogFileWriter: Writing logcat output to content URI: content://com.android.externalstorage.documents/document/primary%3ADownload%2Fk9mail-logs.txt
`

--- a/app/ui/legacy/src/main/java/com/fsck/k9/ui/messageview/MessageViewContainerFragment.kt
+++ b/app/ui/legacy/src/main/java/com/fsck/k9/ui/messageview/MessageViewContainerFragment.kt
@@ -44,12 +44,16 @@ class MessageViewContainerFragment : Fragment() {
     private val messageViewFragment: MessageViewFragment
         get() {
             check(isResumed)
-            val itemId = adapter.getItemId(messageReference)
-
-            // ViewPager2/FragmentStateAdapter don't provide an easy way to get hold of the Fragment for the active
-            // page. So we're using an implementation detail (the fragment tag) to find the fragment.
-            return childFragmentManager.findFragmentByTag("f$itemId") as MessageViewFragment
+            return findMessageViewFragment()
         }
+
+    private fun findMessageViewFragment(): MessageViewFragment {
+        val itemId = adapter.getItemId(messageReference)
+
+        // ViewPager2/FragmentStateAdapter don't provide an easy way to get hold of the Fragment for the active
+        // page. So we're using an implementation detail (the fragment tag) to find the fragment.
+        return childFragmentManager.findFragmentByTag("f$itemId") as MessageViewFragment
+    }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
@@ -221,7 +225,7 @@ class MessageViewContainerFragment : Fragment() {
     }
 
     fun onPendingIntentResult(requestCode: Int, resultCode: Int, data: Intent?) {
-        messageViewFragment.onPendingIntentResult(requestCode, resultCode, data)
+        findMessageViewFragment().onPendingIntentResult(requestCode, resultCode, data)
     }
 
     private class MessageViewContainerAdapter(fragment: Fragment) : FragmentStateAdapter(fragment) {

